optimum/intel/openvino/modeling_base.py

```diff
@@ -0,0 +1,234 @@
+                standard cache should not be used.
+            force_download (`bool`, *optional*, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            file_name(`str`, *optional*):
+                The file name of the model to load. Overwrites the default file name and allows one to load the model
+                with a different name.
+            local_files_only(`bool`, *optional*, defaults to `False`):
+                Whether or not to only look at local files (i.e., do not try to download the model).
+        """
+        from_onnx = kwargs.pop("from_onnx", False)
+        local_files_only = kwargs.pop("local_files_only", False)
+        config_dict = kwargs.pop("config", {})
+        config = PretrainedConfig.from_dict(config_dict)
+        default_file_name = ONNX_WEIGHTS_NAME if from_onnx else OV_WEIGHTS_NAME
+        file_name = file_name or default_file_name
+
+        # Load the model from local directory
+        if os.path.isdir(model_id):
+            file_name = os.path.join(model_id, file_name)
+            bin_file_name = file_name.replace(".xml", ".bin") if not from_onnx else None
+            model = cls.load_model(file_name, bin_file_name)
+            kwargs["model_save_dir"] = model_id
+        # Download the model from the hub
+        else:
+            model_file_names = [file_name]
+            # If not ONNX then OpenVINO IR
+            if not from_onnx:
+                model_file_names.append(file_name.replace(".xml", ".bin"))
+            file_names = []
+            for file_name in model_file_names:
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                file_names.append(Path(model_cache_path).name)
+            kwargs["model_save_dir"] = Path(model_cache_path).parent
+            bin_file_name = file_names[1] if not from_onnx else None
+            model = cls.load_model(file_names[0], bin_file_name=bin_file_name)
+        return cls(model, config=config, **kwargs)
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        save_dir: Union[str, Path] = default_cache_path,
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        **kwargs,
+    ):
+        """
+        Export a vanilla Transformers model into an ONNX model using `transformers.onnx.export_onnx`.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                Directory from which to load
```

Comment:
If I understand correctly: directory from which to load or Transformers model id (e.g. bert-base-uncased)?

Reply:
I updated the docstring to make it clearer